redirect/client.js

```diff
@@ -0,0 +1,54 @@
+(function() {
+
+    function apply(context, template) {
+
+        /**
+         * (window.frameElement || {}).id === "siteEditorFrame" is present in order to prevent redirection from
+         * occurring while in either the Template Editor or Campaign Editor.
+         */
+        const currentPage = window.location.hostname + window.location.pathname.replace(/\/$/, "");
+        const targetPage = context.targetPageUrl.replace(/http(s)?\:\/\//, "");
+        if ((window.frameElement || {}).id === "siteEditorFrame" || currentPage !== targetPage) {
```

Comment:
Add a check for `redirectUrl` here
```suggestion
        if ((window.frameElement || {}).id === "siteEditorFrame" || currentPage !== targetPage || !context.redirectUrl) {
```

Reply:
This would only check that the field isn't empty, but it wouldn't account for the scenario where `window.location.href` is equal to the redirect URL, which needs to be checked so that we don't risk landing in a loop of redirecting to the page we're currently on.

Reply:
@amaskiellEG Doesn't the `currentPage !== targetPage` already account for that? The Promise will run if and only if currentPage === targetPage. So in other words, the Promise won't run if currentPage is equal to the redirect URL.

Reply:
The target page and redirect URL are 2 different things.

Target page is page that's landed on, while the redirect URL is the page to which we're sending someone. While I'd like to think that someone would be smart enough to make sure that this Template wouldn't load on a page to which it's also supposed to redirect, I'd rather not even allow for the possibility of trying to do the redirect if we land on a page that matches the redirect URL (i.e. if the redirect URL is https://www.northerntrailoutfitters.com/default/homepage and we land on that URL, we shouldn't try to redirect to it).

Reply:
If the Promise can only run on pages where `currentPage === targetPage` is true, that means it can't run on the redirectURL. Unless target URL happens to equal redirect URL -- is that the condition you're trying to avoid? If so, then what you're trying to do makes sense.

Reply:
Yeah, that's at least one condition I had in mind. Considering the potential negative impact of an infinite redirect loop, I think it's worth explicitly protecting against it.